### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by leagris** [koalaman/shellcheck#2065]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1035
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
[[ $0 =~. ]]
```

#### Here's what shellcheck currently says:
```
Line 2	SC1035: You are missing a required space here.
```
#### Here's what I wanted or expected to see:

No space required here for Bash Regex Matching

**Comment by koalaman** [comment]
Which version of Bash are you using? I'm getting an error with this snippet on 5.0.16(1)-release.

**Comment by leagris** [comment]
Sorry cannot reproduce now! BASH_VERSION="5.0.17(1)-release"
